packages/mongodb-chatbot-server/src/processors/generateResponseWithSearchTool.test.ts

```diff
@@ -0,0 +1,386 @@
+          ...makeMakeGenerateResponseWithSearchToolArgs(),
+          languageModel: mockThrowingLanguageModel,
+        });
+
+        const result = await generateResponse(generateResponseBaseArgs);
+
+        expect(result.messages[0].role).toBe("user");
+        expect(result.messages[0].content).toBe(latestMessageText);
+        expect(result.messages.at(-1)?.role).toBe("assistant");
+        expect(result.messages.at(-1)?.content).toBe(mockLlmNotWorkingMessage);
+      });
+    });
+
+    describe("streaming mode", () => {
+      // Create a mock DataStreamer implementation
+      const makeMockDataStreamer = () => {
+        const mockStreamData = jest.fn();
+        const mockConnect = jest.fn();
+        const mockDisconnect = jest.fn();
+        const mockStream = jest.fn().mockImplementation(async () => {
+          // Process the stream and return a string result
+          return "Hello";
+        });
+        const dataStreamer = {
+          connected: false,
+          connect: mockConnect,
+          disconnect: mockDisconnect,
+          streamData: mockStreamData,
+          stream: mockStream,
+        } as DataStreamer;
+
+        return dataStreamer;
+      };
+      test("should handle successful streaming", async () => {
+        const mockDataStreamer = makeMockDataStreamer();
+        const generateResponse = makeGenerateResponseWithSearchTool(
+          makeMakeGenerateResponseWithSearchToolArgs()
+        );
+
+        const result = await generateResponse({
+          ...generateResponseBaseArgs,
+          shouldStream: true,
+          dataStreamer: mockDataStreamer,
+        });
+
+        expect(mockDataStreamer.streamData).toHaveBeenCalledTimes(3);
+        expect(mockDataStreamer.streamData).toHaveBeenCalledWith({
+          data: "Final",
+          type: "delta",
+        });
+        expect(mockDataStreamer.streamData).toHaveBeenCalledWith({
+          type: "references",
+          data: expect.any(Array),
+        });
+        expectSuccessfulResult(result);
+      });
+
+      // TODO: (EAI-995): make work as part of guardrail changes
+      test.skip("should handle successful generation with guardrail", async () => {
+        // TODO: add
+      });
+      // TODO: (EAI-995): make work as part of guardrail changes
+      test.skip("should handle streaming with guardrail rejection", async () => {
+        // TODO: add
+      });
+
+      test("should handle error in language model", async () => {
+        const generateResponse = makeGenerateResponseWithSearchTool({
+          ...makeMakeGenerateResponseWithSearchToolArgs(),
+          languageModel: mockThrowingLanguageModel,
+        });
+
+        const dataStreamer = makeMockDataStreamer();
+        const result = await generateResponse({
+          ...generateResponseBaseArgs,
+          shouldStream: true,
+          dataStreamer,
+        });
+
+        // TODO: verify dataStreamer was called
```

Comment:
Need todo still or covered in another PR?

Reply:
hmm not sure, will add here just in case. easy fix.